### PR TITLE
Added CONTRIBUTING file

### PR DESCRIPTION
**Pull request by WilliamLoy**
This file details how to contribute a blog post to the repository.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+2. Before you start working, make sure your local content is up-to-date and
+   merged with the right upstream branch:
+
+      ```bash
+         git remote add upstream git@github.com:rackerlabs/docs-developer-blog.git
+         git checkout master
+         git fetch upstream
+         git checkout -b name-of-your-branch
+         git merge upstream/master
```
Comment (robb-romans) [inline CONTRIBUTING.md:29]: merge upstream/master should come before checkout -b

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+**Note:** If your post has images, you should place the image files within the newly created
+folder.
+
+#### Formatting your post
+
+The post should contain front-matter, an excerpt, and the actual content.
+
+##### Front-matter
+
+Add Jekyll front-matter (or metadata) to the top of the file you created in
```
Comment (robb-romans) [inline CONTRIBUTING.md:48]: remove Jekyll

but this should be handled automatically by an archetype. the writer will run a command to generate the directory and index.md with initial metadata. Check with @codebyravi .

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+#
+authorAvatar: 'https://www.gravatar.com/avatar/<insert hash for your headshot>'
+bio: "<insert a sentence or two about yourself in first or third person>"
+categories:
+    - This Category
+    - That Category
+    - Other Category
+#
+# Use canonical entry if you are republishing a blog from another site, such as
+# your personal blog.  Do  NOT republish without the autor's explicit permission.
```
Comment (robb-romans) [inline CONTRIBUTING.md:72]: s/autor's/author's

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+twitterDescription:
+twitterTitle:
+---
+```
+
+**NOTE:** The "ogImage" _must_ be a fully-qualified URL. If you'd like to use an
+image asset that is being uploaded as part of your blog post, the pattern for the
+final URL is as follows:
+
+  `<rackspace CDN>/<image name>-<sha256 hash>.<image extension>`
```
Comment (robb-romans) [inline CONTRIBUTING.md:94]: I think we can cut this. Have authors put the image files in the page bundle. Let's get away from rackcdn complexity.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+- Salesforce
+- SDK
+- Security
+- SQL Server
+
+If no category fits, use *General*.
+
+If you'd like to use a category that is not in the list, please send an email
+to infodev@rackspace.com. To avoid being flooded with categories, which might apply
+to only one or two blogs, we have automated throttling. However, notify us so
```
Comment (robb-romans) [inline CONTRIBUTING.md:154]: maybe rephrase this because I don't think we really have "automated throttling"

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+that we can discuss your ideas for a new category.
+
+##### Excerpt
+
+Include an excerpt marker after your first paragraph or so to separate the
+preview text that appears on the blog index page from the full article. To do
+so, use the following HTML comment:
+
+```
+The excerpt paragraph, which should give teh reader a taste of what's to come.
```
Comment (robb-romans) [inline CONTRIBUTING.md:164]: s/teh/the

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+<!-- more -->
+
+The rest of your article.
+```
+
+The marker comment ``<!-- more -->`` must be on its own line, starting at
```
Comment (robb-romans) [inline CONTRIBUTING.md:171]: @codebyravi do we still need this? How are we to handle excerpts in Hugo Markdown posts?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+##### Images
+
+**To include images in your blog**, place them  within their respective blog
+directory. Within your blog, use the following markup:
+
+```markdown
+![Alt text here]({% asset_path YYYY-MM-DD-title-of-your-post/filename.png %})
+```
+
+**To wrap text around an image on the right**, use code similar to the following:
```
Comment (robb-romans) [inline CONTRIBUTING.md:183]: this section needs to be revised for Hugo

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+   PR, the changes don't show up in the new staging link.  To force a rebuild,
+   add a blank line to the bottom of your post and commit the change.
+4. When you're satisfied, the InfoDev team will do a quick editorial review
+   of the content and suggest changes, if necessary.
+5. After the content is ready to go, the InfoDev team will publish the blog on
+   the date that you selected.
+
+Thanks again for your interest in contributing!
+
+##### Ideas to post about
```
Comment (robb-romans) [inline CONTRIBUTING.md:210]: I get that this comes from search terms, but it seems quite outdated and not where we want to go. let's get a new list.

File: CONTRIBUTING.md (hunk 2 above)
Comment (WilliamLoy) [reply CONTRIBUTING.md:48]: I chose to comment out this section out for now.

File: CONTRIBUTING.md (hunk 3 above)
Comment (WilliamLoy) [reply CONTRIBUTING.md:72]: Didn't work on this section and it is now commented out but I corrected the typo anyways.

File: CONTRIBUTING.md (hunk 8 above)
Comment (WilliamLoy) [reply CONTRIBUTING.md:183]: I believe this is the syntax {{- $image := resources.Get "YYYY-MM-DD-title-of-your-post/filename.png"-}} @robb-romans or @codebyravi can you confirm?

File: CONTRIBUTING.md (hunk 5 above)
Comment (catlook) [reply CONTRIBUTING.md:155]: The current system definitely has auto-throttling.  You cannot add a category unless a magic threshold of blogs already has that category.  For example, if I want to use "Microsoft" as a category, I have to find ten or so posts and add the category all at once.  If I do them individually, I get an error.

File: CONTRIBUTING.md (hunk 9 above)
Comment (catlook) [reply CONTRIBUTING.md:211]: Pretty sure this is less than six months old.  How frequently would you like it updated?

Comment (catlook) [reply CONTRIBUTING.md:211]: Note: Heather Fergusson says that an agency collects this info for us, supposedly on a quarterly basis.  However, they are currently off-cadence.  I've asked to be added the list to get access to updates whenever they happen.  If I get that access, I'll update the top search terms as often as we get a new set.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+#### Preparing for your blog entry
+
+1. Fork this repo (https://github.com/rackerlabs/technical-blog), then
+   clone your fork.
+
+2. Before you start working, make sure your local content is up-to-date and
+   merged with the right upstream branch:
+
+      ```bash
+         git remote add upstream git@github.com:rackerlabs/docs-developer-blog.git
```
Comment (catlook) [inline CONTRIBUTING.md:25]: guess the upstream link should change to point to the technical-blog repo.

File: CONTRIBUTING.md (hunk 2 above)
Comment (catlook) [reply CONTRIBUTING.md:49]: I'd love to look at the front-matter/metadata options and help decide what we need to have to create the best possible experience for everyone-readers and writers.

File: CONTRIBUTING.md (hunk 4 above)
Comment (catlook) [reply CONTRIBUTING.md:95]: This image affects what displays if you put a link to your blog in Slack, for example.  I doubt Slack knows how to link to an image file in a repo folder.

File: CONTRIBUTING.md (hunk 5 above)
Comment (catlook) [reply CONTRIBUTING.md:155]: If we elect not to enforce category use, lines 152-155 go away.  We can still leave the suggested category list as a recommendation.

File: CONTRIBUTING.md (hunk 10 above)
Comment (robb-romans) [reply CONTRIBUTING.md:25]: Agreed

File: CONTRIBUTING.md (hunk 5 above)
Comment (robb-romans) [reply CONTRIBUTING.md:155]: Hugo has no implementation that I'm aware of for "auto-throttling" of categories. Suggest we review these categories as part of our redesign and information architecture work.

File: CONTRIBUTING.md (hunk 8 above)
Comment (robb-romans) [reply CONTRIBUTING.md:184]: I don't think you need all that, just `./image-filename.png` but I defer to @codebyravi

File: CONTRIBUTING.md (hunk 9 above)
Comment (robb-romans) [reply CONTRIBUTING.md:211]: @catlook Trending search terms is something that we can ponder internally - but - the goal of this section is to ask for what we *want* to be contributed. I don't think we *want* new articles for many of these dated terms e.g. kibana, micron, java ee, mobile world congress, etc.
